apps/app_directed_pickup.c

```diff
@@ -380,15 +393,18 @@ static struct ast_channel *find_by_part(struct ast_channel *chan, const char *pa
 	/* Try a partial channel name search. */
 	pickup_args.name = part;
 	pickup_args.len = strlen(part);
-	target = ast_channel_callback(find_by_name, NULL, &pickup_args, 0);
+
+	target = ast_channel_callback_safe(find_by_name, "channel name, where pickup is possible", pickup_args.name, NULL, &pickup_args, 0);
+
 	if (target) {
 		return target;
 	}
 
 	/* Now try a search for uniqueid. */
 	pickup_args.name = part;
 	pickup_args.len = 0;
-	return ast_channel_callback(find_by_uniqueid, NULL, &pickup_args, 0);
+
+	return ast_channel_callback_safe(find_by_uniqueid, "channel uniqueid, where pickup is possible", pickup_args.name, NULL, &pickup_args, 0);
```

Comment:
The description is identical to line 229. Shouldn't they be different if the goal is to be unique?
Additionally I think automatically include the filename and line number is more robust than hardcoding descriptions. In which case, there should be a macro that injects these automatically.

Reply:
Fixed